test/examples/uct_hello_world.c

```diff
@@ -368,9 +368,7 @@ static ucs_status_t dev_tl_lookup(const cmd_args_t *cmd_args,
                                 ucs_memory_type_names[test_mem_type],
                                 UCT_TL_RESOURCE_DESC_ARG(&tl_resources[tl_index]),
                                 component_attr.md_resources[md_index].md_name);
-                        uct_release_tl_resource_list(tl_resources);
-                        status = UCS_ERR_INVALID_PARAM;
-                        goto close_md;
+                        continue;
```

Comment:
why need to continue? can we support different memtypes with the same transport?

Reply:
> why need to continue?

just to do same as below (when iface initialization fails, we continue looping)

> can we support different memtypes with the same transport?

IB TLs support CUDA (if GPUDirect is enabled) and HOST memory types simultaneously.

Reply:
> do you expect to find more than 1 resource with the same device and tl names?

ah, I see now what I meant
thank you, it seems that code below doesn't need to continue on an error and we have to fail